### PR TITLE
[Feature:InstructorUI] Ability to Upload JSON to make gradeable

### PR DESCRIPTION
**Pull request by IDzyre**
### Please check if the PR fulfills these requirements:

* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant
* [x] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
The Submitty UI only has the ability to manually create gradeables VIA the UI

### What is the new behavior?
image
image

There is now a button to upload a JSON file containing all of the relevant data. 

# Template
```
{
  "title": "Testing Json",
  "instructions_url": "",
  "id": "hw-1-1-1222",
  "type": "Electronic File"/"Checkpoints"/"Numeric",
  "vcs": {
    "repository_type": "submitty-hosted",
    "vcs_path": "path/to/vcs",
    "vcs_subdirectory": "subdirectory"
  },
  "bulk_upload": true/false,
  "team_gradeable": {
    "team_size_max": 3,
    "inherit_from": "gradeable_id",
    "gradeable_teams_read": false,
  },
  "grading_inquiry": {
    "grade_inquiry_per_component_allowed": true/false,
  },
  "ta_grading": true/false,
  "discussion_thread_id": "thread_id",
  "syllabus_bucket": "Homework"
}
```

Dates and other features will be in future PRs
### Other information?
* [x] Add tests
* [x] Review Security
* [x] Add API route
* [x] Write Documentation

### CODE REVIEW

File: site/app/controllers/admin/AdminGradeableController.php [hunk 1]
```diff
@@ -50,6 +50,32 @@ public function editGradeableRequest($gradeable_id, $nav_tab = 0) {
         'electronic_bulk' => 'TA/Instructor will (bulk) upload scanned .pdf for online manual grading'
     ];
 
+    /**
+     * Creates a gradeable based on uploaded JSON data
+     * @return JsonResponse
+     * @AccessControl(role="INSTRUCTOR")
+     * @Route("/api/upload/{_semester}/{_course}", methods={"POST"})
+     * @Route("/courses/upload/{_semester}/{_course}", methods={"POST"})
```
Comment (cjreed121) [inline site/app/controllers/admin/AdminGradeableController.php:58]: On all of our other routes we have `/courses/{_semester}/{_course}/...`. Is there a reason for it to be different that the others? I think it's better to keep it consistent.

**Review by cjreed121 — CHANGES_REQUESTED** [review]
When trying to open the "Upload Gradeable JSON" modal, it did not work and I received the following error in the console:
image

**Comment by cjreed121** [comment]
Also when submitting only ID, title, and type (electronic) I get a server error and a lot of undefined key warnings. This should fail more gracefully.

**Comment by cjreed121** [comment]
> > When trying to open the "Upload Gradeable JSON" modal, it did not work and I received the following error in the console: image
> 
> I'm not quite sure how to reproduce this. Is this happening when you click the 'upload gradeable json' button ?

Yeah as soon as I click that button

**Review by williamjallen — COMMENTED** [review]
The repeated query appears to be an existing defect in the gradeable creation process.  Given that this route is never used by many people at the same time, it is fine to ignore it by adding it to `.performance_warning_ignore.json`.

**Review by cjreed121 — APPROVED** [review]
Code looks good and tested locally

**Comment by IDzyre** [comment]
https://github.com/Submitty/Submitty/pull/10119 should be merged before this one.

**Comment by bmcutler** [comment]
@IDzyre 
Can you please resolve the merge conflicts?  thanks!

**Pull request merged**